src/components/Menu/FilterSelect/index.tsx

```diff
@@ -31,30 +31,19 @@ const convertToRouteHerf = (
   filter: FilterSelectProps,
   targetFilterKey: string
 ) => {
-  const url = filter.url.startsWith('/') ? `file://${filter.url}` : filter.url;
-  let path = new URL(url).pathname;
+  let url = filter.url.startsWith('/') ? `file://${filter.url}` : filter.url;
+  if (url.includes('?')) url = url.slice(0, url.indexOf('?'));
+  if (url.includes('#')) url = url.slice(0, url.indexOf('#'));
 
-  const firstPathSegment = getFirstPathSegment(path);
-  if (firstPathSegment)
-    path = path.replace(`/${firstPathSegment}`, '/[firstPathSegment]');
+  let path = new URL(url).pathname;
 
-  let queryIndex = path.lastIndexOf('/q/');
+  const queryIndex = path.lastIndexOf('/q/');
   if (queryIndex >= 0) {
     path =
       path.substring(0, queryIndex) +
-      path
-        .substring(queryIndex)
-        .replace(`/${filter.filterKey}`, `/[${filter.filterKind}]`);
+      `/q/${filter.filterKind}/${targetFilterKey}`;
   }
-  return {
-    pathname: path,
-    query: {
-      [filter.filterKind]: targetFilterKey,
-      firstPathSegment: multiLibVersionPlatforms.includes(targetFilterKey)
-        ? firstPathSegment
-        : firstPathSegment?.split('-')[0]
-    }
-  };
+  return path;
```

Comment:
It looks like here we are changing the return type of this function but not how we use it.  Is the expected output for this ever expected to be the object that was previously being returned?